packages/mds-provider/api.ts

```diff
@@ -352,49 +405,26 @@ function api(app: express.Express): express.Express {
       event_types: [VEHICLE_EVENTS.trip_start, VEHICLE_EVENTS.trip_end]
     }
 
-    /* istanbul ignore next */
-    function fail(err: Error): void {
-      log.error('/trips failure', err.stack || err).then(() => {
-        res.status(500).send({
-          error: 'internal_failure',
-          error_description: `trips error: ${err.stack}`
-        })
+    try {
+      const { tripIds } = await db.readTripIds(params)
+      const trips = await asEventTrips(tripIds)
+      res.status(200).send({
+        version: PROVIDER_VERSION,
+        data: {
+          trips
+        },
+        links: asJsonApiLinks(req, skip, take, trips.length)
       })
-    }
-
-    if (newSkool) {
-      db.readTrips(params).then((result: ReadTripsResult) => {
-        const { count, trips } = result
-        log.info('read', trips.length, 'trips of', count, 'in', result)
-        res.status(200).send({
-          version: PROVIDER_VERSION,
-          data: {
-            trips
-          },
-          links: asJsonApiLinks(req, skip, take, count)
-        })
+    } catch (err) {
+      res.status(500).send({
+        error: 'internal_failure',
+        error_description: `trips error: ${err.stack}`
```

Comment:
You should never send the full error stack trace back to the client, that's potentially a massive security hole. The stack trace should only ever be output through `mds-logger`, like `log.error(err)`.

Reply:
I think I inherited this, but I will amend.

Reply:
At some point we should move all of the various exception/error types into a common place and standardize the logging/handling. And perhaps add a function that can convert an `Error` into something safe to return as a response.

Reply:
Yeah, `mds-logger` censors latitude and longitude, which is potentially sensitive information.